ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -167,7 +188,7 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
       if (result.isLoading || result.isFetching || result.data === undefined) continue;
 
       for (let i = 0; i < result.data.series.length; i++) {
-        const timeSeries = result.data.series[i];
+        const timeSeries: TimeSeries | undefined = result.data.series[i];
```

Comment:
For some reason, TS had trouble inferring this type after one of the changes below. Explicitly setting it to fix that.